File: servidores/servidor.js
const net = require("net");

const server = net
  .createServer((socket) => {
    socket.on("data", (data) => {
      console.log(data.toString());
    });
    socket.write(">> Recibido");
    socket.end("> Cerrando servidor");
  })
  .on("error", (err) => {
    console.error(err);
  });

server.listen(9898, () => {
  console.log("opened server on", server.address().port);
});

File: servidores/servidor.rb
require 'socket'
server = TCPServer.new('127.0.0.1', 8888)
puts 'Servidor iniciado en 127.0.0.1:8888'

client = server.accept
while (tmp = client.recv(10) and tmp != 'SALIR')
    puts tmp
    client.puts '>> Recibido'
end